### PR TITLE
Doc - _rev warning

### PR DESCRIPTION
**Pull request by sleto-it**
Hi,

1. Only some initial work is done: a warning message is added
2. This is initial work that needs to be reviewed and in case further progressed
3. A Cookbook with info on how to maintain multiple revisions can be considered - and added in this or in a subsequent PR to not have this one wait too much
4. Please do not commit other things or changes in this PR - to not delay review process. Open other PRs instead. This PR is to address a specific issue
5. Duplicating text in different parts of the documentation is not a nice solution imho - suggested when possible to find other approaches i.e. make use of _include_

Thanks,

### CODE REVIEW

File: Documentation/Books/Manual/Appendix/Glossary.md [hunk 1]
```diff
@@ -124,6 +124,28 @@ The document revision is stored in the `_rev` attribute of a document, and is se
...
+an existing document (version) gets removed: for some time, the old version
+of the document plus the removal operation will be on disk.
+
+On disk it is therefore possible that many revisions of the same document
+(as identified by the same `_key` value) exist at the same time.
+
+Howerver even if there might be multiple revisions of the same document (as
+identified by the same `_key` value) somewhere on the disk, **they are not
+accessible**. Once a document was updated or removed successfully, no query
+or other data retrieval operation done by the users will be able to see it
```
Comment (Simran-B) [inline Documentation/Books/Manual/Appendix/Glossary.md]: Don't you see dead documents in dumps by chance?

**Comment by Simran-B** [comment]
1. 👍 
2. I think we should shorten the text(s) above the warning and remove potentially misleading parts:
  - > It is a string value currently containing an integer number

    This is not true anymore, it is a string which looks like garbage but is actually an encoded value (Hybrid Logical Clock something)
  - > Document revisions can be used to conditionally update, replace or delete documents in the database.

    Yes, but that is not the default behavior. At least in AQL, you need to explicitly set the query option `{ ignoreRevs: false }`. It is the default in the HTTP API though, or isn't it? Either way, we should explain in more detail what is meant and if this applies to the HTTP API only, or link to another page which already has an in-depth explanation (we do for AQL, but need to check re HTTP API)
  - > In order to find a particular revision of a document, you need the document handle and the document revision.

    I'm pretty sure this was written with the HTTP API in mind, but sounds like a general statement. It is not necessary in AQL to specify the `_rev` field however. But even in case of the HTTP API, is this statement even correct? Does it have relevance in cluster setups only? Or is it wrong altogether (from a user perspective)?
3. we may want to avoid the term "revision" in other contexts and rather say something like "if you want to maintain multiple **versions** of a document...". We have the blog article about time travelling in graphs, but for a simple document case, I don't know what we should recommend as best practice.
4. 👍 
5. There is already duplicated content in the docs, but we can try to avoid adding more. Includes are supported in Gitbook via Nunjucks templating (`{% include '...' %}`), but there are the following problems:
   - you can not include files outside of the book root folder (e.g. you can't include `../Manual/foo.md` in `Cookbook/bar.md`). This is hardcoded into Gitbook
   - if you put an `.md` file anywhere in a book folder without including it directly in the `SUMMARY.md`, one of our checks will fail (it is supposed to warn you that you probably forgot to add a file to the summary, because it is assumed that the content won't end up in the output otherwise)
   - if you give the file a different extension, then it will not be found because it does not get copied to ppbooks folder (would require a change to the docu build script)

   We have a custom system for includes, namely the DocuBlocks. These work across books.

   However, you must not put relative links in a DocuBlock which gets included multiple times, because it is likely that the pages which include them are located in different folders and at different levels, often resulting in broken links. It would be great if we could rewrite relative links in the DocuBlock python script to fix this problem, but this will take some to time to implement.

   For the purposes of this PR, we can use a DocuBlock. I already tested it in new Jenkins with success.

**Comment by Simran-B** [comment]
@sleto-it I wrote a bit of new text, which should better explain what the revision field is about, but also does not go into much detail about the value anymore.

---

_dan [4:49 PM]_
The only user-facing guarantee we make is that if we revise a document, then `OLD._rev < NEW._rev`. Other than that, they should ignore it.
Or rather, other than that they should treat it as an opaque value.
So they can pass it in as a pre-condition for updates, like you said, essentially "only update if it matches this revision I retrieved earlier".

_jsteemann [4:56 PM]_
Technically speaking, we don't "guarantee" that officially. What we guarantee is that after an update the revision will be _different_, but not that it will be _higher_.

_dan [5:06 PM]_
Oops, that's right. Even more reason to treat it as an opaque value.

---

A DocuBlock is used to show the same text including a warning in two places of the Manual:

- http://web01.arangodb.biz/documentation/improve-revs-desc/Manual/Appendix/Glossary.html#document-revision
- http://web01.arangodb.biz/documentation/improve-revs-desc/Manual/DataModeling/Documents/DocumentAddress.html#document-revision

File: Documentation/Books/stash/DocumentRevisionMVCC.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+As ArangoDB supports MVCC (Multiple Version Concurrency Control),
```
Comment (danielhlarkin) [inline Documentation/Books/stash/DocumentRevisionMVCC.md]: "As ArangoDB uses MVCC (Multiple Version Concurrency Control) internally,"

File: Documentation/Books/stash/DocumentRevisionMVCC.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+A [_Hybrid Logical Clock_](http://www.cse.buffalo.edu/tech-reports/2014-04.pdf)
+is used.
+
+Within one shard it is guaranteed that two different document revisions
+have a different `_rev` string, even if they are written in the same
+millisecond, and that these stamps are ascending.
+
+Note however that different servers in your cluster might have a clock
+skew, and therefore between different shards or even between different
+collections the time stamps are not guaranteed to be comparable.
```
Comment (danielhlarkin) [inline Documentation/Books/stash/DocumentRevisionMVCC.md]: In a single server setup, `_rev` values are guaranteed to be unique across all documents and all collections.

**Comment by sleto-it** [comment]
thanks for your work on this

**Pull request merged**